src/main/java/thaumicinsurgence/item/ItemResource.java

```diff
@@ -0,0 +1,323 @@
+        super.onUpdate(stack, world, entity, par4, par5);
+        if (!entity.worldObj.isRemote && (stack.getItemDamage() == 11 || stack.getItemDamage() == 12)
+                && entity instanceof EntityLivingBase
+                && !((EntityLivingBase) entity).isEntityUndead()
+                && !((EntityLivingBase) entity).isPotionActive(Config.potionTaintPoisonID)
+                && world.rand.nextInt(4321) <= stack.stackSize) {
+            ((EntityLivingBase) entity).addPotionEffect(new PotionEffect(Config.potionTaintPoisonID, 120, 0, false));
+            if (entity instanceof EntityPlayer) {
+                String s = StatCollector.translateToLocal("tc.taint_item_poison")
+                        .replace("%s", "§5§o" + stack.getDisplayName() + "§r");
+                ((EntityPlayer) entity).addChatMessage(new ChatComponentTranslation(s, new Object[0]));
+                InventoryUtils.consumeInventoryItem((EntityPlayer) entity, stack.getItem(), stack.getItemDamage());
+            }
+        } else if (!entity.worldObj.isRemote && stack.getItemDamage() == 15) {
+            int r = world.rand.nextInt(20000);
+            if (stack.hasTagCompound() && stack.stackTagCompound.hasKey("blurb")) {
+                stack.stackTagCompound.removeTag("blurb");
+            }
+
+            if (r < 20) {
+                Aspect aspect = null;
+                switch (world.rand.nextInt(6)) {
+                    case 0:
+                        aspect = Aspect.AIR;
+                        break;
+                    case 1:
+                        aspect = Aspect.EARTH;
+                        break;
+                    case 2:
+                        aspect = Aspect.FIRE;
+                        break;
+                    case 3:
+                        aspect = Aspect.WATER;
+                        break;
+                    case 4:
+                        aspect = Aspect.ORDER;
+                        break;
+                    case 5:
+                        aspect = Aspect.ENTROPY;
+                }
+
+                if (aspect != null) {
+                    EntityAspectOrb orb = new EntityAspectOrb(world, entity.posX, entity.posY, entity.posZ, aspect, 1);
+                    world.spawnEntityInWorld(orb);
+                }
+            } else if (r == 42 && entity instanceof EntityPlayer
+                    && !ResearchManager
+                            .isResearchComplete(((EntityPlayer) entity).getCommandSenderName(), "FOCUSPRIMAL")
+                    && !ResearchManager
+                            .isResearchComplete(((EntityPlayer) entity).getCommandSenderName(), "@FOCUSPRIMAL")) {
+                                ((EntityPlayer) entity).addChatMessage(
+                                        new ChatComponentTranslation(
+                                                "§5§o" + StatCollector.translateToLocal("tc.primalcharm.trigger"),
```

Comment:
Use the chat formatting enum instad of unicode symbols here (the server-side one, there are two)